doc/reference/dune-project/pin.rst

```diff
@@ -0,0 +1,43 @@
+pin
+---
+
+.. warning::
+
+   Dune Package Management is not final yet and the configuration options
+   are subject to change.
+
+
+Pins are package overrides used in the context of package management. They
+allow to fix a package at a specific version which is not affected by the
+package repositories selected.
+
+.. describe:: (pin ...)
+
+   .. versionadded:: 3.14
+
+   Define a package override.
+
+   .. describe:: (url <string>)
+
+      The URL of the package source.
+
```

Comment:
```suggestion
This can be a path to a directory on the local file system or remote git repository. Local paths can be absolute or relative, and may optionally begin with `file://` though this is not necessary. Remote git repository urls must begin with `git+https://`, for example `git+https://github.com/user/repo`.
```